pkg/k8s/provisioner/provisioner_controller.go

```diff
@@ -0,0 +1,128 @@
+	log := r.log.WithValues("request", req)
+	log.V(1).Info("reconciling provisionerclass")
+
+	pc := &v1alpha1.ProvisionerClass{}
+	if err := r.Client.Get(ctx, req.NamespacedName, pc); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	bundles := &v1alpha1.BundleList{}
+	if err := r.Client.List(ctx, bundles, client.InNamespace(req.Namespace)); err != nil {
+		log.Error(err, "failed to list all the bundles in the req.Namespace")
+		return ctrl.Result{}, err
+	}
+
+	filtered := []*v1alpha1.Bundle{}
+	for _, b := range bundles.Items {
+		log.Info("provisioner", "processing bundle name", b.Name)
+		if string(b.Spec.Class) != pc.Name {
+			log.Info("provisioner", "found bundle name that does not reference the current provisioner class", b.Name)
+			continue
+		}
+		filtered = append(filtered, &b)
+	}
+	if len(filtered) == 0 {
+		log.Info("no bundles found specifying the current provisionerclass")
+		return ctrl.Result{}, nil
+	}
+
+	var errors []error
+	for _, bundle := range filtered {
+		log.Info("provisioner", "found bundle name that references the current provisionerclass", bundle.Name)
+		if err := r.ensureBundleVolume(ctx, bundle.GetName()); err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	return ctrl.Result{}, utilerrors.NewAggregate(errors)
+}
+
+// ensureBundleVolume is a helper method responsible for ensuring the bundle's status
+// references an existing volume that will the content stored in the bundle sources'
+// filesystem.
+func (r *Reconciler) ensureBundleVolume(ctx context.Context, bundleName string) error {
+	fresh := &v1alpha1.Bundle{}
+	if err := r.Client.Get(ctx, getNonNamespacedName(bundleName), fresh); err != nil {
+		return err
+	}
+
+	if fresh.Status.Volume != nil {
+		// TODO(tflannag): Check the status is still valid
+		// TODO(tlfannag): Handle case where the object was just updated and requeued
+		return nil
+	}
+
+	// TODO(tflannag): Avoid hardcoding -- likely can find the PVC by label selecting
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: "manifests", Namespace: r.globalNamespace}, pvc); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+	fresh.Status.Volume = &corev1.LocalObjectReference{
+		Name: pvc.GetName(),
+	}
+
+	r.log.Info("volume", "attempting to update the bundle volume", fresh.GetName(), "with pvc name", pvc.GetName())
+	if err := r.Client.Status().Update(ctx, fresh); err != nil {
+		return err
+	}
+	r.log.Info("volume", "bundle status has been updated to point to a volume created", fresh.GetName())
+
+	return nil
+}
+
+// TODO(tflannag): This logic should be abstracted further away as a newVolume
+// type operation.
+// TODO(tflannag): Need a way to avoid run-once operations that cannot be reconciled
+// when state changes, e.g. avoid an InstallPlan-like scenario.
+// TODO(tflannag): Avoid hardcoding the storage requests.
+func (r *Reconciler) createPVC(bundleName string) (*corev1.PersistentVolumeClaim, error) {
```

Comment:
so do we create the PV at all or is it up to the user to allocate a StorageClass or PV that can be claimed via the associated PVC? I like the latter option frankly, if we state that you need a storage solution in-cluster to use rukpak